docs/navbar.js

```diff
@@ -1,77 +1,92 @@
 
 /**
- * Loads the navbar and invokes the callback upon load.
- *
- * @param {Function} callback the callback invoked upon load
+ * Callback function for loading navbar.html that updates the bar based
+ * on what kind of user is looking at it (firebase privileged or not).
+ * @param {Promise<Map<string, Object>>} firebaseDataPromise Promise with data
+ *     for all disasters
+ * @param {?Function} changeDisasterHandler Function invoked when current
+ *     disaster is changed. location.reload is called if null
+ * @param {Promise<any>} privilegedUserPromise Promise with user shapes data *
+ * @param {string} title Title of the page
+ * @return {Promise<void>}
  */
-function loadNavbar(callback) {
-  $('#navbar').load(getUrlUnderDocs('navbar.html'), () => {
-    $('#map-a').prop('href', getUrlUnderDocs(''));
-    $('#manage-layers-a').prop('href', getUrlUnderDocs(MANAGE_LAYERS_PAGE));
-    $('#manage-disaster-a')
-        .prop('href', getUrlUnderDocs(MANAGE_DISASTERS_PAGE));
-    $('#upload-image-collection-a')
-        .prop('href', getUrlUnderDocs(UPLOAD_IMAGE_COLLECTION_PAGE));
-    $('#help-a').prop('href', getHelpUrl());
-    callback();
-  });
+async function loadNavbar(
+    firebaseDataPromise, changeDisasterHandler, privilegedUserPromise, title) {
+  // console.log($('#nav-toggle'));
+  const navToggle = $('#nav-toggle').hide();
+  const navPublic = $('#public').hide();
+  $('#nav-left')
+      .load(
+          getUrlUnderDocs('disaster_picker.html'),
+          () => initializeDisasterPicker(
+              firebaseDataPromise, changeDisasterHandler));
+  $('#nav-title-header').html(title);
+  $('.help-a').prop('href', getHelpUrl());
+  try {
+    console.log('trying');
+    await privilegedUserPromise;
+  } catch (err) {
+    console.log('error');
+    if (err.code === AUTHENTICATION_ERROR_CODE) {
+      navPublic.show();
+    } else {
+      showError(err, 'Error populating navbar. Try refreshing page.');
+    }
+    return;
+  }
+  console.log('succeeded');
+  navToggle.show();
+  console.log($('#nav-toggle').is(':visible'));
+  $('#map-a').prop('href', getUrlUnderDocs(''));
+  $('#manage-layers-a').prop('href', getUrlUnderDocs(MANAGE_LAYERS_PAGE));
+  $('#manage-disaster-a').prop('href', getUrlUnderDocs(MANAGE_DISASTERS_PAGE));
+  $('#upload-image-collection-a')
+      .prop('href', getUrlUnderDocs(UPLOAD_IMAGE_COLLECTION_PAGE));
 }
 
 /**
  * Loads the navbar with a disaster picker.
- * @param {Promise} firebaseAuthPromise Promise that completes when Firebase
- *     login is done
- * @param {string} title Title of page
- * @param {?Function} changeDisasterHandler Function invoked when disaster is
- *     changed. If not specified, reloads page
- * @param {?Promise<Map<string, Object>>} firebaseDataPromise If caller has
- *     already kicked off a fetch of all disasters, pass that Promise in here to
- *     avoid a duplicate fetch
+ * @param {Object} data
```

Comment:
helpful to use the `@param data.firebaseAuthPromise` syntax below data so arguments are documented. Can do that here rather than in body of function. I think you can also directly destructure, just declaring the function
`function loadNavbarWithPicker({firebaseAuthPromise, title...}) {`